app/black_spots/filters.py

```diff
@@ -56,3 +70,49 @@ def filter_effective_at(self, queryset, effective_at_str):
     class Meta:
         model = BlackSpotSet
         fields = ['record_type']
+
+
+class EnforcerAssignmentFilter(GeoFilterSet):
+    """Filter for enforcer assignments"""
+
+    record_type = django_filters.MethodFilter(name='record_type', action='filter_record_type')
+    polygon = django_filters.MethodFilter(name='polygon', action='filter_polygon')
+    polygon_id = django_filters.MethodFilter(name='polygon_id', action='filter_polygon_id')
+    shift_end = django_filters.MethodFilter(name='shift_end', action='filter_shift_end')
+
+    def filter_polygon(self, queryset, geojson):
+        """Method filter for arbitrary polygon, sent in as geojson"""
+        try:
+            poly = GEOSGeometry(geojson)
+        except ValueError as e:
+            raise ParseError(e)
+        if poly.valid:
+            return queryset.filter(geom__intersects=poly)
+        else:
+            raise ParseError('Input polygon must be valid GeoJSON: ' + poly.valid_reason)
+
+    def filter_polygon_id(self, queryset, poly_uuid):
+        """Method filter for containment within the polygon using id"""
+        try:
+            return queryset.filter(geom__intersects=BoundaryPolygon.objects.get(pk=poly_uuid).geom)
+        except ValueError as e:
+            raise ParseError(e)
+        except BoundaryPolygon.DoesNotExist as e:
+            raise NotFound(e)
+
+    def filter_record_type(self, queryset, rt_id):
+        """Method filter for record type"""
+        return queryset.filter(black_spot_set__record_type=rt_id)
+
+    def filter_shift_end(self, queryset, shift_end_str):
```

Comment:
`shift_start` isn't being used for filtering. `shift_end` determines which set of black spots to use, and then the combination of the two will be used for applying the ToDDoW logic. So I don't think it's needed in here, but correct me if I'm wrong.